### DISCUSSION TITLE
Do programming required hardware knowledge?

### DISCUSSION THREAD

**Discussion by naveenalla3000** [community/community#107513] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Do programming required hardware knowledge?

**Comment by SALAUDDIN06** [comment, marked as answer]
yes it needs some knowledge.

**Reply by naveenalla3000** [reply]
yep!

**Comment by naumanch969** [comment]
For beginners, hardware knowledge may not be essential, but as you progress to coding more complex programs and engaging in competitive programming, having a grasp of hardware concepts becomes increasingly valuable. Acquiring basic hardware knowledge can greatly enhance your abilities as a programmer, especially in more advanced coding scenarios.

**Reply by naveenalla3000** [reply]
Exactly!
